03/arr-to-bst.md

```diff
@@ -0,0 +1,221 @@
+    # Arrange
+    arr = []
+    num = 0
+    while num < 100:
+        arr.append(num)
+        num += 1
+
+    # Act
+    result = arr_to_bst(arr)
+
+    # Assert
+    self.assertTrue(self.is_bst(result))
+    self.assertTrue(self.is_balanced_tree(result))
+
+def test_will_return_none_for_empty_list(self):
+    # Arrange
+    arr = []
+
+    # Act
+    result = arr_to_bst(arr)
+
+    # Assert
+    self.assertEqual(result, None)
+```
+
+</details>
+
+## Prompts
+
+<!-- Question 1 -->
+<!-- prettier-ignore-start -->
+### !challenge
+* type: paragraph
+* id: d0a2663c-8528-41b6-8e21-333812b18876
+* title: Ask Clarifying Questions
+* topics: pse
+##### !question
+
+List three or more questions whose answers would clarify the problem statement.
+
+For each question, provide an explanation which includes the effect your decision would have on how you would approach the problem.
+
+##### !end-question
+
+##### !explanation
+
+Here are some example clarifying questions:
+
+1. What kind of data is stored in the Tree Nodes?
+2. Is the array guaranteed to be non-empty?
+
+##### !end-explanation
+
+### !end-challenge
+<!-- prettier-ignore-end -->
+
+<!-- Question 2 -->
+<!-- prettier-ignore-start -->
+
+### !challenge
+* type: code-snippet
+* language: python3.6
+* id: 275e4d6e-accf-4dd3-89c4-9f23729615e8
+* title: Write Unit Tests
+* topics: pse
+##### !question
+
+1. Use the comments provided to write at least two example input/outputs:
+    * Consider at least one nominal and one edge case.
+    * What is the expected output for the given input?
+    * You can use the examples provided in the prompt, or other examples.
+2. Write unit tests for `arr_to_bst` for the nominal and edge cases you identified in the first step.
+
+*Note: Click the **Run Tests** button to save your tests for instructor feedback. No real tests are actually run again your unit tests.*
+
+##### !end-question
+##### !placeholder
+
+```py
+# example input 1:
+# expected output 1:
+
+# example input 2:
+# expected output 2:
+
+def test_nominal_case():
+    # ^rename with meaningful test name
+    # and complete test implementation below
+    pass
+    # arrange
+
+    # act
+
+    # assert
+
+def test_edge_case():
+    # ^rename with meaningful test name
+    # and complete test implementation below
+    pass
+    
+    # arrange
+    
+    # act
+    
+    # assert
+```
+##### !end-placeholder
+
+##### !tests
+
+```py
+import unittest
+
+class TestPython1(unittest.TestCase):
+  def test_always_pass(self):
+    self.assertEqual(1,1)
+```
+
+##### !end-tests
+##### !explanation 
+
+Example tests:
+
+```python
+def test_will_return_balanced_bst_for_odd_lengthed_list(self):
+        # Arrange
+        arr = [5, 10, 15, 20, 25, 30, 35, 40, 45]
+
+        # Act
+        result = arr_to_bst(arr)
+
+        # Assert
+        self.assertEqual(result.val, 25)
+        self.assertTrue(is_bst(result))
+        self.assertTrue(is_balanced_tree(result))
```

Comment:
Students are taught pytest but not unittest in the classroom phase. While I would expect them to be able to read and understand a unittest, I'm not sure writing them should be within scope for this course. 

```suggestion
        assert result.val == 25
        assert is_bst(result)
        assert is_balanced_tree(result)
```

Reply:
updated 🙌🏾